governance/default_test.go

```diff
@@ -560,3 +563,201 @@ func TestWriteGovernance_idxCache(t *testing.T) {
 		t.Errorf("idxCache has wrong value")
 	}
 }
+
+func getTestCouncil() []common.Address {
+	return []common.Address{
+		common.HexToAddress("0x414790CA82C14A8B975cEBd66098c3dA590bf969"), // Node Address for test
+		common.HexToAddress("0x604973C51f6389dF2782E018000c3AC1257dee90"),
+		common.HexToAddress("0x5Ac1689ae5F521B05145C5Cd15a3E8F6ab39Af19"),
+		common.HexToAddress("0x0688CaC68bbF7c1a0faedA109c668a868BEd855E"),
+	}
+}
+
+func getTestRewards() []common.Address {
+	return []common.Address{
+		common.HexToAddress("0x2A35FE72F847aa0B509e4055883aE90c87558AaD"),
+		common.HexToAddress("0xF91B8EBa583C7fa603B400BE17fBaB7629568A4a"),
+		common.HexToAddress("0x240ed27c8bDc9Bb6cA08fa3D239699Fba525d05a"),
+		common.HexToAddress("0x3B980293396Fb0e827929D573e3e42d2EA902502"),
+	}
+}
+
+func getTestVotingPowers(num int) []uint64 {
+	vps := make([]uint64, 0, num)
+	for i := 0; i < num; i++ {
+		vps = append(vps, 1)
+	}
+	return vps
+}
+
+func TestGovernance_HandleGovernanceVote_None_mode(t *testing.T) {
+	// Create ValidatorSet
+	council := getTestCouncil()
+	rewards := getTestRewards()
+
+	blockCounter := common.Big0
+	valSet := validator.NewWeightedCouncil(council, rewards, getTestVotingPowers(len(council)), nil, istanbul.WeightedRandom, 21, 0, 0, nil)
+	gov := getGovernance()
+	gov.nodeAddress.Store(council[len(council)-1])
+
+	votes := make([]GovernanceVote, 0)
+	tally := make([]GovernanceTallyItem, 0)
+
+	proposer := council[0]
+	self := council[len(council)-1]
+	header := &types.Header{}
+
+	//////////////////////////////////////////////////////////////////////////////////////////////////////////
+	// Test for "none" mode
+	header.Number = blockCounter.Add(blockCounter, common.Big1)
+	header.BlockScore = common.Big1
+	gov.AddVote("governance.unitprice", uint64(22000))
+	header.Vote = gov.GetEncodedVote(proposer, blockCounter.Uint64())
+
+	gov.HandleGovernanceVote(valSet, votes, tally, header, proposer, self)
+
+	if _, ok := gov.changeSet.items["governance.unitprice"]; !ok {
+		t.Errorf("Vote had to be applied but it wasn't")
+	}
+
+	//////////////////////////////////////////////////////////////////////////////////////////////////////////
+	// Test for wrong key
+	header.Number = blockCounter.Add(blockCounter, common.Big1)
+	gov.AddVote("governance.non-existent", true)
```

Comment:
@ehnuje You're right. Actually, if `AddVote` returns `false`, the vote can't be added at all and it is already being tested by other test codes. So this test is meaningless and it will be better to be removed from here.

Reply:
Thanks! I understood.